webpack.config.js

```diff
@@ -79,192 +72,154 @@ const base = {
-                'scratch-gui': './src/index.js'
-            },
-            output: {
-                libraryTarget: 'umd',
-                path: path.resolve('dist'),
-                publicPath: `${STATIC_PATH}/`
-            },
-            externals: {
-                'react': 'react',
-                'react-dom': 'react-dom'
+    .addPlugin(new HtmlWebpackPlugin({
+        chunks: ['gui'],
+        template: 'src/playground/index.ejs',
+        title: 'Scratch 3.0 GUI'
+    }))
+    .addPlugin(new HtmlWebpackPlugin({
+        chunks: ['blocksonly'],
+        filename: 'blocks-only.html',
+        template: 'src/playground/index.ejs',
+        title: 'Scratch 3.0 GUI: Blocks Only Example'
+    }))
+    .addPlugin(new HtmlWebpackPlugin({
+        chunks: ['compatibilitytesting'],
+        filename: 'compatibility-testing.html',
+        template: 'src/playground/index.ejs',
+        title: 'Scratch 3.0 GUI: Compatibility Testing'
+    }))
+    .addPlugin(new HtmlWebpackPlugin({
+        chunks: ['player'],
+        filename: 'player.html',
+        template: 'src/playground/index.ejs',
+        title: 'Scratch 3.0 GUI: Player Example'
+    }))
+    .addPlugin(new CopyWebpackPlugin({
+        patterns: [
+            {
+                from: 'static',
+                to: 'static'
             },
-            module: {
-                rules: base.module.rules.concat([
-                    {
-                        test: /\.(svg|png|wav|mp3|gif|jpg)$/,
-                        loader: 'url-loader',
-                        options: {
-                            limit: 2048,
-                            outputPath: 'static/assets/',
-                            publicPath: `${STATIC_PATH}/assets/`
-                        }
-                    }
-                ])
-            },
-            plugins: base.plugins.concat([
-                new CopyWebpackPlugin({
-                    patterns: [
-                        {
-                            from: 'extension-worker.{js,js.map}',
-                            context: 'node_modules/scratch-vm/dist/web',
-                            noErrorOnMissing: true
-                        }
-                    ]
-                }),
-                // Include library JSON files for scratch-desktop to use for downloading
-                new CopyWebpackPlugin({
-                    patterns: [
-                        {
-                            from: 'src/lib/libraries/*.json',
-                            to: 'libraries',
-                            flatten: true
-                        }
-                    ]
-                })
-            ])
-        })) : []
-);
+            {
+                from: 'extensions/**',
+                to: 'static',
+                context: 'src/examples'
+            }
+        ]
+    }));
+
+// Skip building `dist/` unless explicitly requested
+// It roughly doubles build time and isn't needed for `scratch-gui` development
```

Comment:
On my main dev computer, building each target takes about 30 seconds, or about 1 minute if you do both.